integration/popular_docker_hub_images/popular_docker_images.bats

```diff
@@ -11,490 +11,560 @@ source ${BATS_TEST_DIRNAME}/../../lib/common.bash
-	image="tomee"
-	docker run --rm --runtime=$RUNTIME -i $image bash -c "echo -e 'public class CL{public static void main(String[]a){System.out.println(\"KataContainers\");}}' > CL.java"
+	image="docker.io/library/tomee:latest"
+	sudo ctr image pull $image
+	sudo ctr run --rm --runtime=$RUNTIME -t $image foobar bash -c "echo -e 'public class CL{public static void main(String[]a){System.out.println(\"KataContainers\");}}' > CL.java:latest"
 }
 
 @test "[display version] run an instance in a traefik container" {
-	image="traefik"
-	if docker run --rm --runtime=$RUNTIME -i $image traefik --version | grep "EXEC spawning"; then false; else true; fi
+	image="docker.io/library/traefik:latest"
+	sudo ctr image pull $image
+	if sudo ctr run --rm --runtime=$RUNTIME -t $image foobar traefik --version | grep "EXEC spawning"; then false; else true; fi
 }
 
 @test "[teamspeak] run a teamspeak container" {
-	image="teamspeak"
-	docker run --rm --runtime=$RUNTIME -i -p 9987:9987/udp -p 10011:10011 -p 30033:30033 -e TS3SERVER_LICENSE=accept $image sh -c "printf 'Kata Containers'"
+	image="docker.io/library/teamspeak:latest"
+	sudo ctr image pull $image
+	sudo ctr run --rm --runtime=$RUNTIME -t -p 9987:9987/udp -p 10011:10011 -p 30033:30033 --env TS3SERVER_LICENSE=accept $image foobar sh -c "printf 'Kata Containers':latest"
 }
 
 @test "[run application] run an instance in an ubuntu debootstrap container" {
-	image="ubuntu-debootstrap"
-	docker run --rm --runtime=$RUNTIME -i $image sh -c 'if [ -f /etc/bash.bashrc ]; then echo "/etc/bash.bashrc exists"; fi'
+	image="docker.io/library/ubuntu-debootstrap:latest"
+	sudo ctr image pull $image
+	sudo ctr run --rm --runtime=$RUNTIME -t $image foobar sh -c 'if [ -f /etc/bash.bashrc ]; then echo "/etc/bash.bashrc exists"; fi'
 }
 
 @test "[run application] start server in a vault container" {
-	image="vault"
-	docker run --rm --runtime=$RUNTIME -i -e 'VAULT_DEV_ROOT_TOKEN_ID=mytest' $image timeout 10 vault server -dev
+	image="docker.io/library/vault:latest"
+	sudo ctr image pull $image
+	sudo ctr run --rm --runtime=$RUNTIME -t --env 'VAULT_DEV_ROOT_TOKEN_ID=mytest' $image foobar timeout 10 vault server -dev
 }
 
 @test "[perl application] start wordpress container" {
-	image="wordpress"
-	if docker run --rm --runtime=$RUNTIME -i $image perl -e 'print "test\n"' | grep LANG; then false; else true; fi
+	image="docker.io/library/wordpress:latest"
+	sudo ctr image pull $image
+	if sudo ctr run --rm --runtime=$RUNTIME -t $image foobar perl -e 'print "test\n"' | grep LANG; then false; else true; fi
 }
 
 @test "[run application] start zookeeper container" {
-	image="zookeeper"
-	docker run --rm --runtime=$RUNTIME -i $image zkServer.sh
+	image="docker.io/library/zookeeper:latest"
+	sudo ctr image pull $image
+	sudo ctr run --rm --runtime=$RUNTIME -t $image foobar zkServer.sh
 }
 
 teardown() {
-	clean_env
-	docker rmi $image
+	clean_env_ctr
+	sudo ctr image rm $image
 	# Check that processes are not running
 	run check_processes
```

Comment:
@egernst this function is using docker too

Reply:
that one is a pain. I'm not sure how much utility there

Reply:
Perhaps we can *just* check for containerd-shim-kata-v2 in our case? ie

pgrep -f `which containerd-shim-kata-v2` ?

Reply:
Horrible bash applied. PTAL

Reply:
@egernst, switching the check in the pgrep is fine as we only care for `containerd-shim-kata-v2`.
About not having kata-runtime CLI, we still do build and ship kata-runtime in 2.0 and kata-env / kata-check should work (I just tested it locally).

Reply:
That bash script is actually not as bad as you painted it. :-)
LGTM!